crates/oro-manifest/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use derive_builder::Builder;
+use error::Result;
+use oro_semver::{Version, VersionReq};
+use serde::{Deserialize, Serialize};
+
+pub use error::Error;
+
+mod error;
+
+#[derive(Builder, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OroManifest {
+    #[builder(setter(into, strip_option), default)]
+    pub name: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub version: Option<Version>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub description: Option<String>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub homepage: Option<String>,
+
+    #[serde(default, alias = "licence")]
+    #[builder(setter(into, strip_option), default)]
+    pub license: Option<String>,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub browser: bool,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub bugs: Option<Bugs>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub keywords: Vec<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub bin: Option<Bin>,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub author: Option<PersonField>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub contributors: Vec<PersonField>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub files: Vec<String>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub main: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub man: Option<Man>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub directories: Directories,
+
+    #[serde(rename = "type")]
+    #[builder(setter(into, strip_option), default)]
+    pub module_type: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub exports: Option<Exports>,
+
+    #[builder(setter(strip_option), default)]
+    pub imports: Option<Imports>,
+
+    #[builder(setter(strip_option), default)]
+    pub repository: Option<Repository>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub scripts: HashMap<String, String>,
+
+    #[builder(setter(strip_option), default)]
+    pub config: Option<serde_json::Value>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub os: Vec<String>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub cpu: Vec<String>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub private: bool,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub publish_config: HashMap<String, String>,
+
+    // Deps
+    #[serde(default)]
+    #[builder(default)]
+    pub dependencies: HashMap<String, VersionReq>,
```

Comment:
Hm. So, on the note of preserving author order – serde_json has a `preserve_order` feature that switches how `serde_json::Value::Object` types are represented. I think in order to take advantage of that, we'd have to point at a `Value` here, then we'd provide an `Into<DependencyMap>` (or something?)

Reply:
See my comment above.